biojava-structure/src/main/java/org/biojava/nbio/structure/io/mmcif/DownloadChemCompProvider.java

```diff
@@ -264,9 +264,20 @@ public  ChemComp getChemComp(String recordName) {
 		} catch (IOException e) {
 
 			logger.error("Could not parse chemical component file "+filename, e);
-			//e.printStackTrace();
 
 		}
+		finally{
+			// Now close it
+			if(inStream!=null){
+				try {
+					inStream.close();
+				} catch (IOException e) {
+					// This would be weird...
+					logger.error("COULD NOT CLOSE CHEMICAL COMPONENT FILE ->>>>> RESOURCE LEAK RESOURCE LEAK. MAYDAY!!!!!!!!! MAYDAY!!!!");
```

Comment:
Why don't we throw the `IOException` if this ever happens? Or throw another type of Exception with the same message, if it is so weird that it happens.